winch/codegen/src/codegen/context.rs

```diff
@@ -73,7 +73,9 @@ impl<'a> CodeGenContext<'a> {
             // All of our supported architectures use the float registers for vector operations.
             V128 => self.reg_for_class(RegClass::Float, masm),
             Ref(rt) => match rt.heap_type {
-                WasmHeapType::Func => self.reg_for_class(RegClass::Int, masm),
+                WasmHeapType::Func | WasmHeapType::Extern => {
```

Comment:
cc @saulecabrera I added some bare-bones pieces for `Extern` in a few places in this PR to get winch as "supporting" reference-types. Flagging Winch as not supporting reference types ended up being particularly difficult because it ends up disabling nearly all tests one way or another. The goal here was to just not panic on the existence of `externref`, but if you prefer I can try to figure out how to detect this elsewhere and reject it while keeping these panics.

Reply:
This looks good to me, thanks!